src/app/shared/starring.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ *    Copyright 2019 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+import { User } from './swagger';
+
+export function calculateRate(starredUsers: User[], currentUser: User): boolean {
```

Comment:
The name of this function is sort of confusing given the return type -- when I first saw it I was expecting a rate, e.g., a number. 

I see it was already like this and you moved it here for reuse... I still find it confusing. :)